app/src/main/java/com/spascoding/taskycourse/core/data/util/RequestHelper.kt

```diff
@@ -0,0 +1,69 @@
+package com.spascoding.taskycourse.core.data.util
+
+import com.spascoding.taskycourse.core.data.Result
+import com.spascoding.taskycourse.core.domain.DataError
+import org.json.JSONObject
+import retrofit2.HttpException
+import retrofit2.Response
+import java.io.IOException
+
+object RequestHelper {
+    suspend fun <T> performRequest(
+        request: suspend () -> Response<T>,
+        onSuccess: suspend (T) -> Unit = {}
+    ): Result<T, DataError.Remote> {
+        return try {
+            val response = request()
+            if (response.isSuccessful) {
+                val body = response.body()
+                return if (body == null) {
+                    @Suppress("UNCHECKED_CAST")
+                    onSuccess(Unit as T)
+                    Result.Success(Unit)
+                } else {
+                    onSuccess(body)
+                    Result.Success(body)
+                }
+            } else {
+                //How I can return error massage from server and it needs to do this?
+//                val errorResponse = response.errorBody()?.string()
+//                val errorMessage = handleErrorMassage(errorResponse)
+//                Result.Error(TextError(errorMessage))
```

Comment:
Is this clear now? I remember this comment from the last PR

Reply:
Yes, I just leave in code if I need from it in future.